### PR TITLE
Repro of an issue with ChatOllama in LangChain JS.

### PR DESCRIPTION
**Pull request by JayNakrani**
# TL;DR:
This bug happens on LangChain JS version `0.0.146`. The [latest fixes](https://x.com/Hacubu/status/1705591154408833523?s=20) have fixed this issue so upgrading LangChain JS to `0.0.153` fixed issues for me.

# Bug

The ollama server produces a stream of responses, however, the generated chunks do not make their way to the UI through the `/api/chat` endpoint.

### Next App browser network
Gets empty response, and no stream of responses!
Screenshot from 2023-09-17 10-42-20
Screenshot from 2023-09-17 10-42-12

## Curl for same `/api/chat` endpoint

```shell

$ curl -iv \
 'http://localhost:3333/api/chat' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Cache-Control: no-cache' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: text/plain;charset=UTF-8' \
  -H 'Origin: http://localhost:3333' \
  -H 'Pragma: no-cache' \
  -H 'Referer: http://localhost:3333/' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36' \
  -H 'sec-ch-ua: "Not/A)Brand";v="99", "Google Chrome";v="115", "Chromium";v="115"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  --data-raw '{"messages":[{"role":"user","content":"Meaning of name Dhananjay in two sentences or less."}]}' \
  --compressed
*   Trying 127.0.0.1:3333...
* Connected to localhost (127.0.0.1) port 3333 (#0)
> POST /api/chat HTTP/1.1
> Host: localhost:3333
> Accept-Encoding: deflate, gzip, br, zstd
> Accept: */*
> Accept-Language: en-GB,en-US;q=0.9,en;q=0.8
> Cache-Control: no-cache
> Connection: keep-alive
> Content-Type: text/plain;charset=UTF-8
> Origin: http://localhost:3333
> Pragma: no-cache
> Referer: http://localhost:3333/
> Sec-Fetch-Dest: empty
> Sec-Fetch-Mode: cors
> Sec-Fetch-Site: same-origin
> User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36
> sec-ch-ua: "Not/A)Brand";v="99", "Google Chrome";v="115", "Chromium";v="115"
> sec-ch-ua-mobile: ?0
> sec-ch-ua-platform: "Linux"
> Content-Length: 94
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
HTTP/1.1 200 OK
< content-type: text/plain; charset=utf-8
content-type: text/plain; charset=utf-8
< x-experimental-stream-data: false
x-experimental-stream-data: false
< date: Sun, 17 Sep 2023 14:43:16 GMT
date: Sun, 17 Sep 2023 14:43:16 GMT
< vary: Accept-Encoding
vary: Accept-Encoding
< connection: close
connection: close
< transfer-encoding: chunked
transfer-encoding: chunked

< 
* Closing connection 0
```

### Ollama server logs
Ollama server does produce a stream of response. The following logs are after https://github.com/JayNakrani/ollama/pull/1

```
llama server listening at http://127.0.0.1:60116

{"timestamp":1694961640,"level":"INFO","function":"main","line":1443,"message":"HTTP server listening","hostname":"127.0.0.1","port":60116}
{"timestamp":1694961640,"level":"INFO","function":"log_server_request","line":1157,"message":"request","remote_addr":"127.0.0.1","remote_port":41874,"status":200,"method":"HEAD","path":"/","params":{}}
2023/09/17 10:40:40 llama.go:396: llama.cpp server started in 0.403889 seconds
{"timestamp":1694961640,"level":"INFO","function":"log_server_request","line":1157,"message":"request","remote_addr":"127.0.0.1","remote_port":41874,"status":200,"method":"POST","path":"/tokenize","params":{}}
{"timestamp":1694961640,"level":"INFO","function":"log_server_request","line":1157,"message":"request","remote_addr":"127.0.0.1","remote_port":41874,"status":200,"method":"POST","path":"/tokenize","params":{}}
resp: {Model:llama2:7b CreatedAt:2023-09-17 14:40:45.629539847 +0000 UTC Response: Sure Done:false Context:[] TotalDuration:0s LoadDuration:0s PromptEvalCount:0 PromptEvalDuration:0s EvalCount:0 EvalDuration:0s}
resp: {Model:llama2:7b CreatedAt:2023-09-17 14:40:45.992557181 +0000 UTC Response:, Done:false Context:[] TotalDuration:0s LoadDuration:0s PromptEvalCount:0 PromptEvalDuration:0s EvalCount:0 EvalDuration:0s}
...
...
...logs snipped...
...
...
resp: {Model:llama2:7b CreatedAt:2023-09-17 14:41:33.335187767 +0000 UTC Response: spiritual Done:false Context:[] TotalDuration:0s LoadDuration:0s PromptEvalCount:0 PromptEvalDuration:0s EvalCount:0 EvalDuration:0s}
resp: {Model:llama2:7b CreatedAt:2023-09-17 14:41:33.613824529 +0000 UTC Response: growth Done:false Context:[] TotalDuration:0s LoadDuration:0s PromptEvalCount:0 PromptEvalDuration:0s EvalCount:0 EvalDuration:0s}
resp: {Model:llama2:7b CreatedAt:2023-09-17 14:41:33.910584569 +0000 UTC Response:. Done:false Context:[] TotalDuration:0s LoadDuration:0s PromptEvalCount:0 PromptEvalDuration:0s EvalCount:0 EvalDuration:0s}

llama_print_timings:        load time =  4940.86 ms
llama_print_timings:      sample time =   148.82 ms /   174 runs   (    0.86 ms per token,  1169.17 tokens per second)
llama_print_timings: prompt eval time =  4940.72 ms /    31 tokens (  159.38 ms per token,     6.27 tokens per second)
llama_print_timings:        eval time = 48370.60 ms /   173 runs   (  279.60 ms per token,     3.58 tokens per second)
llama_print_timings:       total time = 53526.74 ms
{"timestamp":1694961694,"level":"INFO","function":"log_server_request","line":1157,"message":"request","remote_addr":"127.0.0.1","remote_port":41874,"status":200,"method":"POST","path":"/completion","params":{}}
{"timestamp":1694961694,"level":"INFO","function":"log_server_request","line":1157,"message":"request","remote_addr":"127.0.0.1","remote_port":41876,"status":200,"method":"POST","path":"/tokenize","params":{}}
resp: {Model:llama2:7b CreatedAt:2023-09-17 14:41:34.216088481 +0000 UTC Response: Done:true Context:[29961 25580 29962 12968 29901 16316 292 310 1024 360 5403 273 29926 388 297 1023 25260 470 3109 29889 13 7900 22137 29901 29871 518 29914 25580 29962 29871 18585 29892 306 29915 29881 367 9796 304 1371 29991 450 1024 360 5403 273 29926 388 756 2999 2099 886 322 1677 1144 29892 541 1244 526 1023 1950 6613 800 29901 13 13 29896 29889 512 379 19910 1608 29892 278 1024 360 5403 273 29926 388 338 10723 515 278 317 12190 768 3838 376 29881 5403 29908 6593 376 24447 29908 322 376 1175 388 29908 6593 376 11569 2173 569 1642 739 338 4049 2183 304 12544 6345 964 17173 29891 13175 470 408 263 5829 310 9324 322 620 2638 663 29889 13 29906 29889 512 777 7560 4185 1973 29892 278 1024 360 5403 273 29926 388 338 884 6942 411 278 379 19910 7339 19521 1459 29894 2219 29892 1058 338 24187 287 408 278 7232 397 2073 310 528 557 2034 313 29888 331 262 457 3081 29897 322 278 1136 441 310 6171 1383 4244 29889 450 1024 338 13112 304 6963 17065 886 310 9324 29892 19872 29892 322 20954 14321 29889] TotalDuration:53.940055537s LoadDuration:410.794832ms PromptEvalCount:31 PromptEvalDuration:4.940722s EvalCount:173 EvalDuration:48.370596s}
[GIN] 2023/09/17 - 10:41:34 | 200 | 53.940656481s |       127.0.0.1 | POST     "/api/generate"
```

### CODE REVIEW

**Comment by JayNakrani** [comment]
FWIW, the [example as shown on LangChain docs](https://js.langchain.com/docs/modules/model_io/models/chat/integrations/ollama#usage) also does not work:


### API Route
```TypeScript
import { StreamingTextResponse, Message } from 'ai';
import { ChatOllama } from 'langchain/chat_models/ollama';
import { StringOutputParser } from 'langchain/dist/schema/output_parser';
import { AIMessage, HumanMessage } from 'langchain/schema';

export const runtime = 'edge';

export async function POST(req: Request) {
  const { messages } = await req.json();

  const llm = new ChatOllama({
    baseUrl: 'http://localhost:11434',
    model: 'llama2:7b',
    temperature: 0.12,
  });

  const stream = await llm
    .pipe(new StringOutputParser())
    .stream((messages as Message[]).map(m =>
      m.role == 'user'
        ? new HumanMessage(m.content)
        : new AIMessage(m.content),
    ));

  return new StreamingTextResponse(stream);
}
```

### Curl logs
```sh
$ curl \
  -iv \
  'http://localhost:3000/api/chat' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  -H 'Cache-Control: no-cache' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: text/plain;charset=UTF-8' \
  -H 'Origin: http://localhost:3000' \
  -H 'Pragma: no-cache' \
  -H 'Referer: http://localhost:3000/chat' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36' \
  -H 'sec-ch-ua: "Not/A)Brand";v="99", "Google Chrome";v="115", "Chromium";v="115"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  --data-raw '{"messages":[{"role":"user","content":"Meaning of name Dhananjay in 5 words or less."}]}' \
  --compressed
*   Trying 127.0.0.1:3000...
* Connected to localhost (127.0.0.1) port 3000 (#0)
> POST /api/chat HTTP/1.1
> Host: localhost:3000
> Accept-Encoding: deflate, gzip, br, zstd
> Accept: */*
> Accept-Language: en-GB,en-US;q=0.9,en;q=0.8
> Cache-Control: no-cache
> Connection: keep-alive
> Content-Type: text/plain;charset=UTF-8
> Origin: http://localhost:3000
> Pragma: no-cache
> Referer: http://localhost:3000/chat
> Sec-Fetch-Dest: empty
> Sec-Fetch-Mode: cors
> Sec-Fetch-Site: same-origin
> User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36
> sec-ch-ua: "Not/A)Brand";v="99", "Google Chrome";v="115", "Chromium";v="115"
> sec-ch-ua-mobile: ?0
> sec-ch-ua-platform: "Linux"
> Content-Length: 88
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
HTTP/1.1 200 OK
< content-type: text/plain; charset=utf-8
content-type: text/plain; charset=utf-8
< date: Sun, 17 Sep 2023 15:13:53 GMT
date: Sun, 17 Sep 2023 15:13:53 GMT
< x-experimental-stream-data: false
x-experimental-stream-data: false
< Vary: Accept-Encoding
Vary: Accept-Encoding
< Connection: keep-alive
Connection: keep-alive
< Keep-Alive: timeout=5
Keep-Alive: timeout=5
< Transfer-Encoding: chunked
Transfer-Encoding: chunked

< 
* Connection #0 to host localhost left intact
```

### Ollama server logs
With logging https://github.com/JayNakrani/ollama/pull/1

```
llama server listening at http://127.0.0.1:64663

{"timestamp":1694963598,"level":"INFO","function":"main","line":1443,"message":"HTTP server listening","hostname":"127.0.0.1","port":64663}
{"timestamp":1694963599,"level":"INFO","function":"log_server_request","line":1157,"message":"request","remote_addr":"127.0.0.1","remote_port":59352,"status":200,"method":"HEAD","path":"/","params":{}}
2023/09/17 11:13:19 llama.go:396: llama.cpp server started in 2.401545 seconds
{"timestamp":1694963599,"level":"INFO","function":"log_server_request","line":1157,"message":"request","remote_addr":"127.0.0.1","remote_port":59352,"status":200,"method":"POST","path":"/tokenize","params":{}}
{"timestamp":1694963599,"level":"INFO","function":"log_server_request","line":1157,"message":"request","remote_addr":"127.0.0.1","remote_port":59352,"status":200,"method":"POST","path":"/tokenize","params":{}}
...
...
...snipped...
...
...
resp: {Model:llama2:7b CreatedAt:2023-09-17 15:13:53.033929408 +0000 UTC Response: Le Done:false Context:[] TotalDuration:0s LoadDuration:0s PromptEvalCount:0 PromptEvalDuration:0s EvalCount:0 EvalDuration:0s}
resp: {Model:llama2:7b CreatedAt:2023-09-17 15:13:53.355937601 +0000 UTC Response:ader Done:false Context:[] TotalDuration:0s LoadDuration:0s PromptEvalCount:0 PromptEvalDuration:0s EvalCount:0 EvalDuration:0s}

llama_print_timings:        load time =  7447.85 ms
llama_print_timings:      sample time =    41.73 ms /    47 runs   (    0.89 ms per token,  1126.21 tokens per second)
llama_print_timings: prompt eval time =  7447.73 ms /    32 tokens (  232.74 ms per token,     4.30 tokens per second)
llama_print_timings:        eval time = 27058.34 ms /    46 runs   (  588.22 ms per token,     1.70 tokens per second)
llama_print_timings:       total time = 34572.35 ms
{"timestamp":1694963633,"level":"INFO","function":"log_server_request","line":1157,"message":"request","remote_addr":"127.0.0.1","remote_port":59352,"status":200,"method":"POST","path":"/completion","params":{}}
{"timestamp":1694963633,"level":"INFO","function":"log_server_request","line":1157,"message":"request","remote_addr":"127.0.0.1","remote_port":59354,"status":200,"method":"POST","path":"/tokenize","params":{}}
resp: {Model:llama2:7b CreatedAt:2023-09-17 15:13:53.634613892 +0000 UTC Response: Done:true Context:[29961 25580 29962 518 25580 29962 16316 292 310 1024 360 5403 273 29926 388 297 29871 29945 3838 470 3109 29889 518 29914 25580 29962 518 29914 25580 29962 29871 2266 526 29871 29945 3838 393 8453 278 6593 310 278 1024 360 5403 273 29926 388 29901 13 13 29896 29889 3159 9347 296 13 29906 29889 3767 549 13 29941 29889 10811 1693 13 29946 29889 1913 2966 2738 13 29945 29889 951 1664] TotalDuration:36.996976561s LoadDuration:2.407668055s PromptEvalCount:32 PromptEvalDuration:7.447733s EvalCount:46 EvalDuration:27.058343s}
[GIN] 2023/09/17 - 11:13:53 | 200 | 36.997774299s |       127.0.0.1 | POST     "/api/generate"
```
